frontend/src/component/common/NewConstraintAccordion/NewConstraintAccordionList/NewConstraintAccordionList.tsx

```diff
@@ -0,0 +1,264 @@
+                                        rel='noopener noreferrer'
+                                    >
+                                        <StyledHelp />
+                                    </a>
+                                </StyledHelpWrapper>
+                            </StyledAddCustomLabel>
+                            <Button
+                                type='button'
+                                onClick={onAdd}
+                                variant='outlined'
+                                color='primary'
+                                data-testid='ADD_CONSTRAINT_BUTTON'
+                            >
+                                Add constraint
+                            </Button>
+                        </div>
+                    }
+                />
+            </StyledContainer>
+        );
+    },
+);
+
+interface IConstraintList {
+    constraints: IConstraint[];
+    setConstraints?: React.Dispatch<React.SetStateAction<IConstraint[]>>;
+    state: IUseWeakMap<IConstraint, IConstraintAccordionListItemState>;
+}
+
+export const NewConstraintAccordionList = forwardRef<
+    IConstraintAccordionListRef | undefined,
+    IConstraintList
+>(({ constraints, setConstraints, state }, ref) => {
+    const { context } = useUnleashContext();
+
+    const onEdit =
+        setConstraints &&
+        ((constraint: IConstraint) => {
+            state.set(constraint, { editing: true });
+        });
+
+    const onRemove =
+        setConstraints &&
+        ((index: number) => {
+            const constraint = constraints[index];
+            state.set(constraint, {});
+            setConstraints(
+                produce((draft) => {
+                    draft.splice(index, 1);
+                }),
+            );
+        });
+
+    const onSave =
+        setConstraints &&
+        ((index: number, constraint: IConstraint) => {
+            state.set(constraint, {});
+            setConstraints(
+                produce((draft) => {
+                    draft[index] = constraint;
+                }),
+            );
+        });
+
+    const onAutoSave =
+        setConstraints &&
+        ((index: number, constraint: IConstraint) => {
+            state.set(constraint, { editing: true });
+            setConstraints(
+                produce((draft) => {
+                    draft[index] = constraint;
+                }),
+            );
+        });
+
+    const onCancel = (index: number) => {
+        const constraint = constraints[index];
+        state.get(constraint)?.new && onRemove?.(index);
+        state.set(constraint, {});
+    };
+
+    if (context.length === 0) {
+        return null;
+    }
+
+    return (
+        <StyledContainer id={constraintAccordionListId}>
+            {constraints.map((constraint, index) => (
+                // biome-ignore lint: reason=objectId would change every time values change - this is no different than using index
+                <Fragment key={index}>
```

Comment:
This was changed. Previously we used `objectId(constraint)` but this objectId would change every time the values of the object were updated, which did not provide a stable reference. This caused the animation to re-trigger in edit mode whenever we re-rendered the component because React thought it was a new object when the id changed. Since it's not stable anyway, and we don't have any stable id for each constraint we might as well resort to index here.